src/matcha_ml/services/global_config_service.py

```diff
@@ -0,0 +1,139 @@
+"""Global config service for creating and modifying a users global config files."""
+import os
+import uuid
+from typing import Any, Dict, Optional
+
+import yaml
+
+from matcha_ml.errors import MatchaPermissionError
+
+
+class GlobalConfigurationService:
+    """A Global Config Service for interacting and updating a users global config file."""
+
+    _instance: Optional["GlobalConfigurationService"] = None
+    _user_id: Optional[str] = None
+    _analytics_opt_out: bool = False
+
+    def __new__(cls) -> "GlobalConfigurationService":
+        """Singleton class definition.
+
+        Returns:
+            GlobalConfigurationService: Already existing initialised object, otherwise a new singleton object
+        """
+        if cls._instance is None:
+            cls._instance = super().__new__(cls)
+            # Check if config.yaml file exists and read in variables to the class
+            if os.path.exists(str(cls._instance.default_config_file_path)):
```

Comment:
`str(cls._instance.default_config_file_path)` is being used in multiple places. Can we not create a variable and reuse it on line 27, 29 and 33.